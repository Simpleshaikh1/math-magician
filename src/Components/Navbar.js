import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Link, useLocation } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: '/calculator', text: 'Calculator' },
  { path: '/quotes', text: 'Quotes' },
];

const Navbar = () => {
  const location = useLocation();

  const isActive = (path) => (location.pathname === path ? { color: 'purple' } : {});

  return (
    <nav>
      <ul>
        {links.map((link) => (
          <li key={link.text}>
            <Link to={link.path} style={isActive(link.path)}>
              {link.text}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
