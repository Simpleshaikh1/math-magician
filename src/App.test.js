import { BrowserRouter, Routes, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../src/Components/Navbar';
import Calculator from '../src/Logic/calculate';
import Qoutes from '../src/Routes/displayQoute';
import Home from '../src/Routes/Home';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <>
          <Navbar />
          <Routes>
            <Route path="/calculator" element={<Calculator />} />
            <Route path="/quotes" element={<Qoutes />} />
            <Route path="/" element={<Home />} />
          </Routes>
        </>
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
