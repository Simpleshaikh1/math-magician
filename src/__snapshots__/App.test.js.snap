// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "purple",
            }
          }
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
          style={Object {}}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quotes"
          onClick={[Function]}
          style={Object {}}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="home-page"
  >
    <h1>
       Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>,
]
`;
